Integrate deal making with Filecoin via Singularity

masih opened the pull request:
Integrate start-up with Singularity and configuration to make deals with Filecoin storage providers.

Update README, and add example env files.

Fixes: https://github.com/filecoin-project/motion/issues/113

masih committed:
Integrate deal making with Filecoin via Singularity

Integrate start-up with Singularity and configuration to make deals with
Filecoin storage providers.

Update README, and add example env files.

Fixes: https://github.com/filecoin-project/motion/issues/113

masih commented:
@willscott @hannahhoward thoughts on what to do with the CI test in this repo? Do we want to set up devnet here again or should we just disable it?

willscott commented:
I think having the integration CI running in the motion repo is the right place - that's where more changes will be and where we want to verify that the changes don't break s3 functionality.

masih committed:
Move integration tests to `motion` repo

Remove integration tests from here and move to `motion` repo, in order
to:
* reduce duplicate CI setup, since motion repo already stands up a
devnet for testing.
* make sure the connector remains compatible with the latest commit to
main of Motion repo.

See:
 - https://github.com/filecoin-project/motion/pull/117

masih committed:
Add Check docker compose CI job

Add CI job for dry run of docker compose to catch syntax errors.

xinaxu commented on docker-compose.yaml:
```diff
@@ -1,14 +1,127 @@
+        condition: service_healthy
+
+  singularity_content_provider:
+    image: ghcr.io/data-preservation-programs/singularity:${SINGULARITY_VERSION:-0.3.2}
```
Is it the right version

masih replied:
I think it is the one that works with Motion 0.2.0?

masih replied:
Going ahead with this version until there is a new tagged version of motion that's tested with singularity.

masih replied:
Version to be updated once [this](https://github.com/filecoin-project/motion/issues/147) is sorted out. Cc @gammazero
